bundles/io/org.openhab.io.gcal/src/main/java/org/openhab/io/gcal/internal/GCalEventDownloader.java

```diff
@@ -533,4 +607,134 @@ public void updated(Dictionary<String, ?> config) throws ConfigurationException
+            if (StringUtils.isEmpty(userdata)) {
+                tokenPath = new File("etc");
+            } else {
+                tokenPath = new File(userdata);
+            }
+
+            File tokenFile = new File(tokenPath, TOKEN_PATH);
+
+            FileDataStoreFactory fileDataStoreFactory = new FileDataStoreFactory(tokenFile);
+            DataStore<StoredCredential> datastore = fileDataStoreFactory.getDataStore("gcal_oauth2_token");
+
+            credential = loadCredential("openhab", datastore);
+            if (credential == null) {
+
+                GenericUrl genericUrl = new GenericUrl("https://accounts.google.com/o/oauth2/device/code");
+
+                Map<String, String> mapData = new HashMap<String, String>();
+                mapData.put("client_id", client_id);
+                mapData.put("scope", CalendarScopes.CALENDAR);
+                UrlEncodedContent content = new UrlEncodedContent(mapData);
+                HttpRequestFactory requestFactory = HTTP_TRANSPORT.createRequestFactory(new HttpRequestInitializer() {
+                    @Override
+                    public void initialize(HttpRequest request) {
+                        request.setParser(new JsonObjectParser(JSON_FACTORY));
+                    }
+                });
+                HttpRequest postRequest = requestFactory.buildPostRequest(genericUrl, content);
+
+                Device device = postRequest.execute().parseAs(Device.class);
+                // no access token/secret specified so display the authorisation URL in the log
+                logger.info(
+                        "################################################################################################");
+                logger.info("# Google-Integration: U S E R   I N T E R A C T I O N   R E Q U I R E D !!");
+                logger.info("# 1. Open URL '{}'", device.verification_url);
+                logger.info("# 2. Type provided code {} ", device.user_code);
+                logger.info("# 3. Grant openHAB access to your Google calendar");
+                logger.info(
+                        "# 4. openHAB will automatically detect the permiossions and complete the authentication process");
+                logger.info("# NOTE: You will only have {} mins before openHAB gives up waiting for the access!!!",
+                        device.expires_in);
+                logger.info(
+                        "################################################################################################");
+
+                logger.debug("Got access code");
+                logger.debug("user code :" + device.user_code);
+                logger.debug("device code :" + device.device_code);
+                logger.debug("expires in:" + device.expires_in);
+                logger.debug("interval :" + device.interval);
+                logger.debug("verification_url :" + device.verification_url);
```

Comment:
A small speed improve for the above block would be to enclose it in `if (logger.isDebugEnabled())`.